v1/server/server.go

```diff
@@ -2569,6 +2581,43 @@ func (s *Server) makeRego(_ context.Context,
 	return rego.New(opts...), nil
 }
 
+func stringPathToQuery(urlPath string) (ast.Body, error) {
+	ref, err := stringPathToDataRef(urlPath)
+	if err != nil {
+		return nil, err
+	}
+
+	return parseRefQuery(ref.String())
+}
+
+// parseRefQuery parses a string into a query ast.Body.
+// The query must be comprised of a single ref, or an error will be returned.
+func parseRefQuery(str string) (ast.Body, error) {
```

Comment:
Arguably, this check might be enough query sanitization; as we're checking that whatever we parsed, it contains exactly one ref, and nothing else.